pymdownx/magiclink.py

```diff
@@ -84,10 +94,26 @@ def handleMatch(self, m):
 class MagiclinkExtension(Extension):
     """Add Easylink extension to Markdown class."""
 
+    def __init__(self, *args, **kwargs):
+        """Initialize."""
+
+        self.config = {
+            'hide_protocol': [
+                False,
+                "If 'True', links are displayed without the initial ftp://, http:// or https://"
+                "- Default: False"
+            ]
+        }
+        super(MagiclinkExtension, self).__init__(*args, **kwargs)
+
     def extendMarkdown(self, md, md_globals):
         """Add support for turning html links and emails to link tags."""
 
-        md.inlinePatterns.add("magic-link", MagiclinkPattern(RE_LINK, md), "<not_strong")
+        config = self.getConfigs()
```

Comment:
What I was really looking for was to attach the config to the MagiclinkPattern object and let it determine when to strip protocol by accessing self.config as was done in the last iteration (just minus all the passing in of the arguments, super, and popping the argument off the param dict).  `el.set("href", self.sanitize_url(self.unescape(href.strip())))` should occur last (even if the stripping won't ever actually affect the sanitizing).

```python
magic_link = MagiclinkPattern(RE_LINK, md)
magic_link.config = self.getConfigs()
md.inlinePatterns.add('magic-link", magic_link, "<not_strong")
```

I don't see it necessary to derive a new class to append 2-3 lines of code, especially when that code is coming after sanitizing when it should come before.  The final URL should be passed in to be sanitized.